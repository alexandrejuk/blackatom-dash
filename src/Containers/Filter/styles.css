.cardCollapsed > .ant-card-body {
  display: none;
}
.cartTitle > * {
  margin-left: 10px;
}
.ant-card-body {
  display: flex;
  flex-wrap: wrap;
}
.input-group {
  padding: 0 5px;
}
.label-filter {
  display: block
}