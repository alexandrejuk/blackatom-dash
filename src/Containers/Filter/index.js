import React, { PureComponent } from 'react'
import {
  Card,
  DatePicker,
  Input,
  Button
} from 'antd'
import classnames from 'classnames'
import './styles.css'

const { RangePicker } = DatePicker

const FilterHeader = ({ onShowMore }) => <div>
  <RangePicker />
  <Input style={{ width: "200px"}}/>
  <Button onClick={onShowMore}> Mais </Button>
</div>

class Filter extends PureComponent {
  state = {
    collapsed: false,
  }

  handleOnShowMore = () => {
    console.log(this.state.collapsed)
    this.setState((state) => ({
      collapsed: !state.collapsed
    }))
  }

  render () {
    const cardClasses = classnames({
      cardCollapsed: this.state.collapsed
    })

    return (<div>
      <Card
        title={<FilterHeader onShowMore={this.handleOnShowMore}/>}
        className={cardClasses}
        style={{ width: '100%' }}
      >
        <p>Card content</p>
        <p>Card content</p>
        <p>Card content</p>
      </Card>
    </div>)
  }
} 

export default Filter