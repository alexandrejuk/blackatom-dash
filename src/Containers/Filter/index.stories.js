import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Filter from '.'

const filters = [
  { property: 'name', type: 'text', label: 'Name', width: "500px" },
  { property: 'age', type: 'number', label: 'Age', width: "200px" },
  { property: 'gender', type: 'text', label: 'Gender', width: "200px" },
]

storiesOf('Filter', module)
  .add('filter default', () => <Filter filters={filters} />)
