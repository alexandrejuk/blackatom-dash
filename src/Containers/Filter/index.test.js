import React from 'react'
import Filter from '.'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

const filters = [
  { property: 'name', type: 'text', label: 'Name', width: "500px" },
  { property: 'age', type: 'number', label: 'Age', width: "200px" },
  { property: 'gender', type: 'text', label: 'Gender', width: "200px" }
]

let wrapper = null
beforeAll(() => {
  wrapper = shallow(<Filter filters={filters} />)
})

test('Filter should render correctly', () => {
  const component = renderer.create(<Filter filters={filters} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Filter should have three Inputs', () => {
  const filter = wrapper.find('Input')
  expect(filter.length).toBe(3)
})

test('Filter should have these property in Input', () => {
  const filter = wrapper.find('Input')

  for(let i =0; i < filter.length; i++) {
    const inputProps = filter.at(i).props()
    expect(inputProps.name).toBe(filters[i].property)
    expect(inputProps.type).toBe(filters[i].type)
  }
})

