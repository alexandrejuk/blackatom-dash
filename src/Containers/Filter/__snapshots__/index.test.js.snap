// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly 1`] = `
<div>
  <div
    className="ant-card ant-card-bordered"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          <div
            className="cartTitle"
          >
            <span
              className="ant-calendar-picker"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
              tabIndex={0}
            >
              <span
                className="ant-calendar-picker-input ant-input"
                onClick={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
              >
                <input
                  className="ant-calendar-range-picker-input"
                  placeholder="Start date"
                  readOnly={true}
                  tabIndex={-1}
                  value=""
                />
                <span
                  className="ant-calendar-range-picker-separator"
                >
                   ~ 
                </span>
                <input
                  className="ant-calendar-range-picker-input"
                  placeholder="End date"
                  readOnly={true}
                  tabIndex={-1}
                  value=""
                />
                <i
                  className="anticon anticon-calendar ant-calendar-picker-icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="calendar"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </i>
              </span>
            </span>
            <input
              className="ant-input"
              disabled={false}
              onKeyDown={[Function]}
              style={
                Object {
                  "width": "200px",
                }
              }
              type="text"
            />
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="anticon anticon-up"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="up"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                  />
                </svg>
              </i>
              <span>
                 Mais Filtros 
              </span>
            </button>
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <span>
                 Limpar Filtros 
              </span>
            </button>
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <span>
                 Pesquisar 
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="input-group"
      >
        <label
          className="label-filter"
        >
          Name
           
        </label>
        <input
          className="ant-input"
          disabled={false}
          name="name"
          onKeyDown={[Function]}
          style={
            Object {
              "width": "500px",
            }
          }
          type="text"
        />
      </div>
      <div
        className="input-group"
      >
        <label
          className="label-filter"
        >
          Age
           
        </label>
        <input
          className="ant-input"
          disabled={false}
          name="age"
          onKeyDown={[Function]}
          style={
            Object {
              "width": "200px",
            }
          }
          type="number"
        />
      </div>
      <div
        className="input-group"
      >
        <label
          className="label-filter"
        >
          Gender
           
        </label>
        <input
          className="ant-input"
          disabled={false}
          name="gender"
          onKeyDown={[Function]}
          style={
            Object {
              "width": "200px",
            }
          }
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
